### PR TITLE
Port to API V8.

### PR DESCRIPTION
**Pull request by jchristgit**
This is based on the official changelog, found at
https://discord.com/developers/docs/change-log#api-and-gateway-v8.

The following user-facing changes are introduced:

- Updating the permission overwrite `type` type and documentation. Code using
  permission overwrites may need changing from the literal `"role"` and
  `"user"` wordings to the new number-based format.
- Remove `game` field usages in the presence cache in favour of `activities`.
- Remove `embed_enabled` and `embed_channel_id` from `Nostrum.Struct.Guild`
  (`widget_enabled` and `widget_channel_id`, the documented alternatives, are
  already present).

### CODE REVIEW

**Comment by jchristgit** [comment]
This needs some more testing, I forgot a local stash here with the ratelimiter.

**Comment by jchristgit** [comment]
Thank you pixel

File: lib/nostrum/api/ratelimiter.ex [hunk 1]
```diff
@@ -110,11 +110,13 @@ defmodule Nostrum.Api.Ratelimiter do
   end
 
   defp wait_for_timeout(request, timeout, from) do
+    truncated = :erlang.trunc(timeout)
```
Comment (jb3) [inline lib/nostrum/api/ratelimiter.ex:113]:
```suggestion
    truncated = :erlang.ceil(timeout)
```

**Review by jb3 — CHANGES_REQUESTED** [review]

**Review by jb3 — APPROVED** [review]

**Comment by jchristgit** [comment]
Thanks for the review 🙂

**Pull request merged**